### ISSUE TITLE
Can't Start the Miner in Windows Error 0xc000007b

### ISSUE THREAD

**Issue by jch9678** [RavenCommunity/kawpowminer#25]
I created a .bat with --help and got this error.  Any ideas?
Thanks

---------------------------
kawpowminer.exe - Application Error
---------------------------
The application was unable to start correctly (0xc000007b). Click OK to close the application. 
---------------------------
OK   
---------------------------

**Comment by jch9678** [comment]
I've tried updating .NET runtimes (even Core) and installed the latest C++ redistributables.

**Comment by nyakze** [comment]
AFAIK this is AMD issue with kpm and pretty common.
Can you provide more details:
Miner build version
OS version and build number (use winver)
Driver version

**Comment by jch9678** [comment]
Sorry for the tardy reply, you are correct I am using AMD on Windows

Miner build version: 1.2.2
OS version and build number (use winver): Version 1809 (OS Build 17763.973)
Driver version: 19.2
gpu model: Vega 56 (reference model)

**Comment by nyakze** [comment]
Tested on same OS Build, but with driver version 20.2.2, gpu rx480 - working fine.

**Comment by tale01** [comment]
Window 10 Version 1909 (OS Build 18363.815)
AMD driver 20.2.2, gpu R9 NANO - get "The application was unable to start correctly (0xc000007b). Click OK to close the application."

I think driver version is not root cause.

If kawpowminer can run on Win 10 Version 1809 (OS Build 17763.973) with Driver 20.2.2 only, it very specify for all AMD miner.

**Comment by nyakze** [comment]
Then the issue gotta lie somewhere else. Not driver and not windows build.

**Comment by tale01** [comment]
Hard to analysis. I have 4 PC diffence Win 10 OS bulid. 
It can run only 1 PC. (Win 10 Version 1903 (OS Build 18362.778))
I check .NET runtimes installed,but is the same version. XD

**Comment by tale01** [comment]
This is not root cause. 
2 PC has the same install. PC 1 can run miner and PC 2 cann't run miner.
![MVC](https://user-images.githubusercontent.com/31344227/81580848-26b77e00-93d8-11ea-82d8-4f0ba7cc8cbe.png)

**Comment by jch9678** [comment]
1909 doesn't work on several machines.

1809 doesn't work - The code execution cannot proceed because VCRUNTIME140_1.dll was not found. Reinstalling the program may fix this problem.     

I encountered this problem on the first machine but installed the latest .NET runtimes and it solved that one but the current problem presented itself

1903 doesn't work - but different OS build than Tale01 18362.592

**Comment by jch9678** [comment]
@tale01 do you have Visual Studio installed on both machines?

**Comment by tale01** [comment]
> @tale01 do you have Visual Studio installed on both machines?

Yes

**Comment by nyakze** [comment]
VCRUNTIME140_1.dll issue is fixed by installing https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads

**Comment by jch9678** [comment]
Yes, that's what I did when I encountered that particular error.

**Comment by tale01** [comment]
Compare 2 PC.  Same Windows version, MSVC++ and AMD driver but OS bulid difference.
But I think not root cause.

![minning](https://user-images.githubusercontent.com/31344227/81712890-5da49700-949f-11ea-868b-b877af0bbe76.png)

![dontmining](https://user-images.githubusercontent.com/31344227/81712909-639a7800-949f-11ea-8711-ee93619a3cda.jpg)

**Comment by tale01** [comment]
Compared. What the cause???
![minning](https://user-images.githubusercontent.com/31344227/81713550-38fcef00-94a0-11ea-9bf3-0928b2712d78.png)

![dontmining](https://user-images.githubusercontent.com/31344227/81713557-3bf7df80-94a0-11ea-9e43-8972f82c231c.jpg)

**Comment by jch9678** [comment]
Here are the differences. 

Different builds for version 1809 - Works 17763.1039 vs 17763.1158 Doesn't Work

Different .NET Core Runtime versions - Works 2.2.1 vs 2.2.8 Doesn't Work

Redistributables:
2008 x64 is not installed on the working machine
2012 x86 and x64 are not installed on the working machine
2013 x86 and x64 are installed on both machines
2015-2019 x64 are installed on machines
2015 x86 14.0.24123 is only installed on the working machine 

I think this last one is the key. 
Can someone tell us what version of Visual Studio this binary was compiled on?

**Comment by tale01** [comment]
Tested 2015 x86 14.0.24123 is not a key

**Comment by jch9678** [comment]
Do you want to try .NET Core 2.2.1 on your non-working machine?

**Comment by tale01** [comment]
Tried. .NET Core 2.2.1 & 2.2.8 not a key

**Comment by jeroz1** [comment]
Miner on telegram reports the same issue

Windows 10; rx580 4gb
![F65B3ECB-0D6F-4CCD-96D5-143A7D073943](https://user-images.githubusercontent.com/44138303/82147217-84bbe980-984e-11ea-9777-278c9840a8ba.jpeg)

**Comment by ParthibanPeriyasamy18** [comment]
Any Solution for this error?  

I tried everything above suggested.  No Help.

**Comment by jch9678** [comment]
I never found one, I had to move on.

**Comment by Bolle-VVout** [comment]
Did you found "VCRUNTIME140_1.dll" if not get it from above link. If you have maybe you can copy it to the same dir as your miner, did help me a while ago with some miner I'm not sure any more witch.

It has to be some .dll problem if I read this.
https://wikifixes.com/en/errors/dll/0xc000007b.dll/?_sid=QDFOtg12vl

Last thing I can think off is that it's searching for cuda files anyway: I know you're running AMD's but don't know exactly what builds you're using and how the initialisation of kawpowminer demands cuda libs.

**Comment by ankah2000** [comment]
Any one else having this issue? I did most (all I think) of the recommended above and could not fix it. Running Vegas 64 here. Could it be a Vega issue?

**Comment by Pabrollo** [comment]
Same error here, running RX 590. I tried pretty much all the steps here.

**Comment by VoldurX** [comment]
Going to add to it. 

Getting the same problem with RX 580. Driver version 20.9.2

**Comment by nyakze** [comment]
As there is no other solution, I suggest using other miners for now (unfortunately all alternatives are closed source and have dev fee). Here's a list: https://raven.wiki/w/Mining#Miners

**Comment by Roman20202020** [comment]
Sorry for the google translator. installed new RX580 4GB.
I suffered with this error for two days, reinstalled .NET and C ++ and what I just didn’t do with dll.
I took them out of the motherboard. I installed an old Geforce 550 ti. Installed the drivers. rebooted and put back RX580. and it all worked.

**Comment by tidux** [comment]
Same issue present on 20H2 Pro.

**Comment by r0manux** [comment]
Hello, same issue, installed Visual C++ Redistributable Runtimes All-in-One from TechPowerUp without succes.
Running Sapphire RX570 8GB on Windows 10 Pro x64, v2004, build 19041.867.
Maybe the attached log from Dependency Walker helps the devs.
GL
[kawpowminer.exe-DependencyWalker-profiler-log.txt](https://github.com/RavenCommunity/kawpowminer/files/6188408/kawpowminer.exe-DependencyWalker-profiler-log.txt)

**Comment by seperatis** [comment]
Same issue; has anyone found fix yet?

Windows 10 PRO

AMD Radeon RX 580 GPU

**Comment by wimjan123** [comment]
Yeah i got the same issue too, a fix would be awesome!

Windows 10

AMD RX 5600XT
I7 8700K

**Comment by LeonVerde420** [comment]
Same issue here. 
Windows 10
RX580 4gb Nitro+

**Comment by ariwerth** [comment]
Still an issue apparently:
Windows 10
RX580
